libcudacxx/include/cuda/std/__cccl/attributes.h

```diff
@@ -67,6 +67,12 @@
 #  define _CCCL_NO_UNIQUE_ADDRESS
 #endif
 
+// Passing objects with nested [[no_unique_address]] to kernels leads to data corruption
+// This happens up to clang18
+#if defined(_CCCL_NO_UNIQUE_ADDRESS) && defined(_CCCL_COMPILER_CLANG)
+#  define _CCCL_HAS_NO_ATTRIBUTE_NO_UNIQUE_ADDRESS
```

Comment:
`_CCCL_HAS_NO_ATTRIBUTE_NO_UNIQUE_ADDRESS` is currently only used in mdspan, so that is the minimal place to fix

Reply:
> ```c++
> #if defined(_CCCL_NO_UNIQUE_ADDRESS)
> ```

`_CCCL_NO_UNIQUE_ADDRESS` is always defined. maybe this should be:

```c++
#if !defined(_CCCL_HAS_NO_ATTRIBUTE_NO_UNIQUE_ADDRESS) && defined(_CCCL_COMPILER_CLANG)
```

for the record, i've run into problems with `[[no_unique_address]]` on clang, gcc, nvcc, and nvc++, probably msvc too. almost everyplace i've ever used it, i've had to remove it.